### PR TITLE
[server] don't start DB if already running, fix post test db cleanup

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Don't start the DB (incl. migrations) if it is already running In server's `yarn test` 

### <samp>🤖 Generated by Copilot at 8be44b5</samp>

Refactored and renamed several test files for the `AuthProviderEntry` entity and the `gitpod-db` module. Extracted a common `resetDB` function to delete all entities except the builtin users before each test. Updated the `package.json` file of the `components/server` module to optimize the test database initialization. Fixed the import paths of the `resetDB` function in some test files of the `components/server` module.

</details>

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/src/project-db.spec.db.ts [hunk 1]
```diff
@@ -30,9 +29,7 @@ class ProjectDBSpec {
 
     async wipeRepo() {
         const typeorm = testContainer.get<TypeORM>(TypeORM);
-        const manager = await typeorm.getConnection();
-        await manager.getRepository(DBProject).delete({});
-        await manager.getRepository(DBUser).delete({});
+        await resetDB(typeorm);
```
Comment (svenefftinge) [inline components/gitpod-db/src/project-db.spec.db.ts:32]: Extracting all post-test DB cleanups to one method that ensures we don't delete needed state (e.g. admin user)

**Review by AlexTugarev — APPROVED** [review]
LGTM!

**Pull request merged**